docs/using-connect/integration.mdx

```diff
@@ -0,0 +1,377 @@
+		oracleMetrics,
+	)
+	app.SetPrepareProposal(proposalHandler.PrepareProposalHandler())
+	app.SetProcessProposal(proposalHandler.ProcessProposalHandler())
+}
+```
+
+Next, set up the `PreBlocker`. This involves:
+
+- **Aggregate Function:** Setting the aggregator function that combines all reported prices into one final price per currency pair.
+- **Currency Pair Strategy:** Setting the currency pair strategy. For this example, we will use the `DeltaCurrencyPairStrategy` which encodes/decodes the price as the difference between the current price and the previous price. While other strategies are available, we recommend this one for most applications.
+- **Data Compression Codecs:** Setting the compression strategy for vote extensions and extended commits.
+
+```go oracle.go
+package app
+
+import (
+    oracleclient "github.com/skip-mev/slinky/service/clients/oracle"
+    servicemetrics "github.com/skip-mev/slinky/service/metrics"
+
+    oraclepreblock "github.com/skip-mev/slinky/abci/preblock/oracle"
+    compression "github.com/skip-mev/slinky/abci/strategies/codec"
+    "github.com/skip-mev/slinky/abci/strategies/currencypair"
+    "github.com/skip-mev/slinky/pkg/math/voteweighted"
+)
+
+func (app *App) initializeABCIExtensions(oracleClient oracleclient.OracleClient, oracleMetrics servicemetrics.Metrics) {
+	// ... 
+
+	// Create the aggregation function that will be used to aggregate oracle data
+	// from each validator.
+	aggregatorFn := voteweighted.MedianFromContext(
+		app.Logger(),
+		app.StakingKeeper,
+		voteweighted.DefaultPowerThreshold,
+	)
+	veCodec := compression.NewCompressionVoteExtensionCodec(
+		compression.NewDefaultVoteExtensionCodec(),
+		compression.NewZLibCompressor(),
+	)
+	ecCodec := compression.NewCompressionExtendedCommitCodec(
+		compression.NewDefaultExtendedCommitCodec(),
+		compression.NewZStdCompressor(),
+	)
+
+	// Create the pre-finalize block hook that will be used to apply oracle data
+	// to the state before any transactions are executed (in finalize block).
+	oraclePreBlockHandler := oraclepreblock.NewOraclePreBlockHandler(
+		app.Logger(),
+		aggregatorFn,
+		app.OracleKeeper,
+		oracleMetrics,
+		currencypair.NewDeltaCurrencyPairStrategy(app.OracleKeeper), // IMPORTANT: always construct new currency pair strategy objects when functions require them as arguments.
+		veCodec,
+		ecCodec,
+	)
+
+	app.SetPreBlocker(oraclePreBlockHandler.WrappedPreBlocker(app.ModuleManager))
+}
+```
+
+Next, configure the vote extensions using the vote extension codec, extended commit codec, and aggregator function from the previous step.
+
+```go oracle.go
+package app
+
+import (
+	"time"
+
+	oracleclient "github.com/skip-mev/slinky/service/clients/oracle"
+    servicemetrics "github.com/skip-mev/slinky/service/metrics"
+
+	"github.com/skip-mev/slinky/abci/ve"
+	"github.com/skip-mev/slinky/abci/strategies/currencypair"
+	"github.com/skip-mev/slinky/abci/strategies/aggregator"
```

Comment:
```suggestion

	oracleclient "github.com/skip-mev/slinky/service/clients/oracle"
        servicemetrics "github.com/skip-mev/slinky/service/metrics"

	"github.com/skip-mev/slinky/abci/ve"
	"github.com/skip-mev/slinky/abci/strategies/currencypair"
	"github.com/skip-mev/slinky/abci/strategies/aggregator"
```

Just some code formatting stuff here and there.  Idk if it even shows up in the rendered doc